app/hotels/src/navigation/allHotels/SearchResultsContext.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+
+import * as React from 'react';
+
+export type ResultType = 'list' | 'map';
+
+const defaultValue = {
+  show: 'list',
+  actions: {
+    setResultType: () => {},
+  },
+};
+
+const { Consumer, Provider: ContextProvider } = React.createContext(
+  defaultValue,
+);
+
+type Props = {|
+  +children: React.Node,
+|};
+
+type State = {
+  show: ResultType,
+  +setResultType: (show: ResultType) => void,
+};
+
+class Provider extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      show: 'list',
+      setResultType: this.setResultType,
+    };
+  }
+
+  setResultType = (show: ResultType) => {
+    this.setState({ show });
+  };
+
+  render = () => (
+    <ContextProvider value={this.state}>{this.props.children}</ContextProvider>
+  );
+}
+
+export default { Consumer, Provider };
+
+export function withSearchResultsContext(select: (state: State) => Object) {
```

Comment:
We should create some utility function that converts our contexts into these HOC `withContext`, because there are many like this in the project!

Reply:
Feel free to make PR with it